### PR TITLE
Improve consistency/completeness of datasource documentation

### PR DESCRIPTION
**Pull request by yrodiere**
See the commit messages for a list of improvements.

Best reviewed commit by commit, since there are a few refactorings that mess up the diff.

### CODE REVIEW

File: docs/src/main/asciidoc/datasource.adoc [hunk 1]
```diff
@@ -44,17 +44,17 @@ If you want a better understanding of how all this works, this guide has a lot m
 === Zero Config Setup (Dev Services)
 
 When testing or running in dev mode Quarkus can even provide you with a zero config database out of the box, a feature
-we refer to as Dev Services. Depending on your database type you may need docker installed in order to use this feature. Dev Services
-is supported for the following open source databases:
+we refer to as Dev Services. Depending on your database type you may need docker installed in order to use this feature.
```
Comment (gsmet) [inline docs/src/main/asciidoc/datasource.adoc:47]:
```suggestion
we refer to as Dev Services. Depending on your database type you may need Docker installed in order to use this feature.
```

File: docs/src/main/asciidoc/datasource.adoc [hunk 2]
```diff
@@ -44,17 +44,17 @@ If you want a better understanding of how all this works, this guide has a lot m
...
 * DB2 (container) (requires license acceptance)
-* MSSQL (container) (requires license acceptance)
-
+* Derby (in-process)
+* H2 (in-process)
+* MariaDB (container)
+* Microsoft SQL Server (container) (requires license acceptance)
+* MySQL (container)
+* Oracle Express Edition (container)
+* Postgresql (container)
```
Comment (gsmet) [inline docs/src/main/asciidoc/datasource.adoc:57]:
```suggestion
* PostgreSQL (container)
```

File: docs/src/main/asciidoc/datasource.adoc [hunk 3]
```diff
@@ -653,6 +653,14 @@ with GraalVM native-images.
 The lack of JMX support is a natural consequence of running in native and is unlikely to be resolved.
 The integration with OCI could be resolved, if you need it we suggest opening a support request with the MySQL Connector/J maintainers.
 
+=== Oracle
+
+`jdbc:oracle:driver_type:[username/password]@database_specifier`
```
Comment (gsmet) [inline docs/src/main/asciidoc/datasource.adoc:658]: I would drop the `[username/password]` part given we don't want it to be used like this.

`database_specifier` contains the host and port?

**Review by gsmet — CHANGES_REQUESTED** [review]
Added a few small suggestions while we're at it + one question.

**Comment by yrodiere** [comment]
Thanks. I applied your suggestions and answered your question.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Thanks!

**Pull request merged**